<div align="right" style=" font-size: 80%; text-align: center; margin: 0 auto">
<img src="https://raw.githubusercontent.com/Explore-AI/Pictures/master/ExploreAI_logos/Logo blue_dark.png"  style="width:25px" align="right";/>
</div>

# Logical and comparison operators ii
© ExploreAI Academy

> ⚠️ This notebook will not run on Google Colab because it cannot connect to a local database. Please make sure that this notebook is running on the same local machine as your MySQL Workbench installation and MySQL `united_nations` database.

## Learning objectives

In this train we will learn:
- How to use the IS NULL and IS NOT NULL statements
- How to use the IS IN and IS NOT IN statements
- Understand if there is any correlation between the GDP and availibility of drinking water and sanitation services in Sub-Saharan Africa

## Connecting to our MySQL database

Using our `Access_to_Basic_Services` table in our `united_nations` database we created in MySQL Workbench, we want to answer some questions about our dataset. We can apply the same queries we used in MySQL Workbench in this notebook if we connect to our MySQL server by running the cells below.


In [21]:
# Load and activate the SQL extension to allows us to execute SQL in a Jupyter notebook. 
# If you get an error here, make sure that mysql and pymysql is installed correctly. 

%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [22]:
# Establish a connection to the local database using the '%sql' magic command,
# Replace 'password' with our connection password and `db_name` with our database name. 
# If you get an error here, please make sure the database name or password is correct.

%sql mysql+pymysql://root:password@localhost:3306/united_nations


To make a query, we add the `%%sql` command to the start of a cell, create one open line and then the query like below and run the cell.

In [23]:
%%sql

SELECT 
    *
FROM
    Access_to_Basic_Services
LIMIT 5;

 * mysql+pymysql://root:***@localhost:3306/united_nations
5 rows affected.


[('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2015, Decimal('94.67'), Decimal('98.00'), Decimal('17.542806'), Decimal('184.39'), Decimal('2699700.00'), Decimal('4.93')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2016, Decimal('94.67'), Decimal('98.00'), Decimal('17.794055'), Decimal('137.28'), Decimal('2699700.00'), Decimal('4.96')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2017, Decimal('95.00'), Decimal('98.00'), Decimal('18.037776'), Decimal('166.81'), Decimal('2699700.00'), Decimal('4.90')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2018, Decimal('95.00'), Decimal('98.00'), Decimal('18.276452'), Decimal('179.34'), Decimal('2699700.00'), Decimal('4.85')),
 ('Central and Southern Asia', 'Central Asia', 'Kazakhstan', 2019, Decimal('95.00'), Decimal('98.00'), Decimal('18.513673'), Decimal('181.67'), Decimal('2699700.00'), Decimal('4.80'))]

## Exercise


We will be working with the `united_nations.Access_to_Basic_Services` table, which contains information about different countries, their access to basic services, and their estimated GDP.

In this exercise We would like to determine if the GDP of a country, specifically in Sub-Saharan Africa has any correlation to its access to basic services


### Task 1 
Select data from the Sub-Saharan African region during the year 2020.

In [32]:
%%sql
SELECT *
FROM united_nations.Access_to_Basic_Services
WHERE time_period = 2020 and region = 'Sub-Saharan Africa'
limit 10;

 * mysql+pymysql://root:***@localhost:3306/united_nations
10 rows affected.


[('Sub-Saharan Africa', 'Eastern Africa', 'Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('12.220227'), Decimal('2.65'), Decimal('25680.00'), Decimal('1.03')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('1.090156'), Decimal('3.18'), Decimal('23180.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('117.190911'), Decimal('107.66'), Decimal('1128571.26'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('51.985780'), Decimal('100.67'), Decimal('569140.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('28.225177'), Decimal('13.05'), Decimal('581800.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('19.377061'), Decimal('12.18'), Decimal('94280.00'), Decimal('0.91')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('1.265740'), Decimal('11.40'), Decimal('2030.00'), Decimal('8.63')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Mayotte', 2020, Decimal('96.00'), Decimal('100.00'), None, None, None, None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('31.178239'), Decimal('14.03'), Decimal('786380.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('13.146362'), Decimal('10.18'), Decimal('24670.00'), Decimal('11.83'))]

### Task 2

Sometimes there are null values in our entries. Any country having Null values for their GDP should not be included in our query as they will not help us determine if there is any correlation between GDP and access to basic services. For this task determine if there are any NULL values in the GDP column

In [41]:
%%sql
# Add your code here
SELECT *
FROM united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND  
    Est_gdp_in_billions is null


 * mysql+pymysql://root:***@localhost:3306/united_nations
9 rows affected.


[('Sub-Saharan Africa', 'Eastern Africa', 'Mayotte', 2020, Decimal('96.00'), Decimal('100.00'), None, None, None, None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Réunion', 2020, Decimal('100.00'), Decimal('100.00'), None, None, None, None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'South Sudan', 2020, Decimal('48.33'), Decimal('22.33'), Decimal('10.606227'), None, Decimal('631930.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'United Republic of Tanzania', 2020, Decimal('65.00'), Decimal('34.00'), None, None, None, None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Congo', 2020, Decimal('69.00'), Decimal('17.67'), None, None, None, None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Democratic Republic of the Congo', 2020, Decimal('47.67'), Decimal('15.33'), None, None, None, None),
 ('Sub-Saharan Africa', 'Western Africa', "Côte d'Ivoire", 2020, Decimal('70.67'), Decimal('34.67'), None, None, None, None),
 ('Sub-Saharan Africa', 'Western Africa', 'Gambia', 2020, Decimal('79.33'), Decimal('44.33'), None, None, None, None),
 ('Sub-Saharan Africa', 'Western Africa', 'Saint Helena', 2020, Decimal('99.00'), Decimal('100.00'), None, None, None, None)]

### Task 3

If there are any Null values exclude them from your query.

In [40]:
%%sql
# Add your code here
SELECT *
FROM united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND  
    Est_gdp_in_billions is not null


 * mysql+pymysql://root:***@localhost:3306/united_nations
38 rows affected.


[('Sub-Saharan Africa', 'Eastern Africa', 'Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('12.220227'), Decimal('2.65'), Decimal('25680.00'), Decimal('1.03')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('1.090156'), Decimal('3.18'), Decimal('23180.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('117.190911'), Decimal('107.66'), Decimal('1128571.26'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('51.985780'), Decimal('100.67'), Decimal('569140.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('28.225177'), Decimal('13.05'), Decimal('581800.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('19.377061'), Decimal('12.18'), Decimal('94280.00'), Decimal('0.91')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('1.265740'), Decimal('11.40'), Decimal('2030.00'), Decimal('8.63')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('31.178239'), Decimal('14.03'), Decimal('786380.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('13.146362'), Decimal('10.18'), Decimal('24670.00'), Decimal('11.83')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('16.537016'), Decimal('6.88'), Decimal('627340.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('44.404611'), Decimal('37.60'), Decimal('200520.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.927715'), Decimal('18.11'), Decimal('743390.00'), Decimal('6.03')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('15.669666'), Decimal('21.51'), Decimal('386850.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('33.428486'), Decimal('53.62'), Decimal('1246700.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('26.491087'), Decimal('40.77'), Decimal('472710.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('5.343020'), Decimal('2.33'), Decimal('622980.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('16.644701'), Decimal('10.72'), Decimal('1259200.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('2.292573'), Decimal('15.31'), Decimal('257670.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.218641'), Decimal('0.47'), Decimal('960.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('2.546402'), Decimal('14.93'), Decimal('566730.00'), Decimal('21.02')),
 ('Sub-Saharan Africa', 'Southern Africa', 'Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('1.180655'), Decimal('3.98'), Decimal('17200.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.254100'), Decimal('2.23'), Decimal('30360.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('2.489098'), Decimal('10.58'), Decimal('823290.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'South Africa', 2020, Decimal('92.00'), Decimal('78.67'), Decimal('58.801927'), Decimal('337.62'), Decimal

### Task 4

Lets get an idea if there's any correlation between the GDP and access to basic services for the top 5 economies in Sub-Saharan Africr. The top 5 GDP's are : ('Nigeria','South Africa','Ethiopia','Kenya','Ghana'). Make sure your query only includes these countries.

In [43]:
%%sql
# Add your code here
SELECT *
FROM united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND  
    Est_gdp_in_billions is not null
AND 
    country_name in ('Nigeria','South Africa','Ethiopia','Kenya','Ghana')

 * mysql+pymysql://root:***@localhost:3306/united_nations
5 rows affected.


[('Sub-Saharan Africa', 'Eastern Africa', 'Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('117.190911'), Decimal('107.66'), Decimal('1128571.26'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('51.985780'), Decimal('100.67'), Decimal('569140.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'South Africa', 2020, Decimal('92.00'), Decimal('78.67'), Decimal('58.801927'), Decimal('337.62'), Decimal('1213090.00'), Decimal('24.34')),
 ('Sub-Saharan Africa', 'Western Africa', 'Ghana', 2020, Decimal('84.67'), Decimal('23.00'), Decimal('32.180401'), Decimal('70.04'), Decimal('227533.00'), None),
 ('Sub-Saharan Africa', 'Western Africa', 'Nigeria', 2020, Decimal('77.33'), Decimal('42.67'), Decimal('208.327405'), Decimal('432.20'), Decimal('910770.00'), None)]

Based on your results. Is there any correlation between the GDP and access to basic services? Does a higher GDP translate to better services?

### Task 5

We only looked at 5 countries in the previous query. Lets have a look at the rest of Sub-Saharan Africa .
Exclude the countries mentioned in the previous task

In [47]:
%%sql
# Add your code here

SELECT *
FROM united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND  
    Est_gdp_in_billions is not null
AND 
    country_name not in ('Nigeria','South Africa','Ethiopia','Kenya','Ghana')

 * mysql+pymysql://root:***@localhost:3306/united_nations
33 rows affected.


[('Sub-Saharan Africa', 'Eastern Africa', 'Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('12.220227'), Decimal('2.65'), Decimal('25680.00'), Decimal('1.03')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('1.090156'), Decimal('3.18'), Decimal('23180.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('28.225177'), Decimal('13.05'), Decimal('581800.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('19.377061'), Decimal('12.18'), Decimal('94280.00'), Decimal('0.91')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('1.265740'), Decimal('11.40'), Decimal('2030.00'), Decimal('8.63')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('31.178239'), Decimal('14.03'), Decimal('786380.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('13.146362'), Decimal('10.18'), Decimal('24670.00'), Decimal('11.83')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('16.537016'), Decimal('6.88'), Decimal('627340.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('44.404611'), Decimal('37.60'), Decimal('200520.00'), None),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.927715'), Decimal('18.11'), Decimal('743390.00'), Decimal('6.03')),
 ('Sub-Saharan Africa', 'Eastern Africa', 'Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('15.669666'), Decimal('21.51'), Decimal('386850.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('33.428486'), Decimal('53.62'), Decimal('1246700.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('26.491087'), Decimal('40.77'), Decimal('472710.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('5.343020'), Decimal('2.33'), Decimal('622980.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('16.644701'), Decimal('10.72'), Decimal('1259200.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('2.292573'), Decimal('15.31'), Decimal('257670.00'), None),
 ('Sub-Saharan Africa', 'Middle Africa', 'Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.218641'), Decimal('0.47'), Decimal('960.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('2.546402'), Decimal('14.93'), Decimal('566730.00'), Decimal('21.02')),
 ('Sub-Saharan Africa', 'Southern Africa', 'Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('1.180655'), Decimal('3.98'), Decimal('17200.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.254100'), Decimal('2.23'), Decimal('30360.00'), None),
 ('Sub-Saharan Africa', 'Southern Africa', 'Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('2.489098'), Decimal('10.58'), Decimal('823290.00'), None),
 ('Sub-Saharan Africa', 'Western Africa', 'Benin', 2020, Decimal('65.33'), Decimal('17.33'), Decimal('12.643123'), Decimal('15.65'), Decimal('112760.00'), None),
 ('Sub-Saharan Africa', 'Western Africa', 'Burkina Faso', 2020, Decimal('53.33'), Decimal('25.00'), Decimal('21.522626'), Decimal('17.93'), Decimal('273600.00'), None),
 ('Sub-Saharan Africa', 'Western Africa', 'Cabo Verde', 2020, Decimal('87.33'), Decimal('78.00'), Decimal('0.582640'), Decimal('1.70'), Decimal('4030

Again. Do you see any correlation? Arrange the `Pct_managed_drinking_water_services` in descending order. Do the countries with the highest percentage to `Pct_managed_drinking_water_services` have a higher GDP then the countries with a lower percentage of `Pct_managed_drinking_water_services` ?
 Now arrange the `Pct_managed_sanitation_services` in descending order. Do countries with a higher `Pct_managed_sanitation_services` also have a higher GDP?

## Solutions

### Task 1

In [19]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
    Time_period = 2020;

 * mysql+pymysql://root:***@localhost:3306/united_nations
47 rows affected.


[('Burundi', 2020, Decimal('70.33'), Decimal('44.33'), Decimal('12.220227'), Decimal('2.65')),
 ('Djibouti', 2020, Decimal('69.00'), Decimal('56.00'), Decimal('1.090156'), Decimal('3.18')),
 ('Ethiopia', 2020, Decimal('58.00'), Decimal('11.67'), Decimal('117.190911'), Decimal('107.66')),
 ('Kenya', 2020, Decimal('67.00'), Decimal('33.67'), Decimal('51.985780'), Decimal('100.67')),
 ('Madagascar', 2020, Decimal('56.33'), Decimal('13.00'), Decimal('28.225177'), Decimal('13.05')),
 ('Malawi', 2020, Decimal('74.33'), Decimal('28.67'), Decimal('19.377061'), Decimal('12.18')),
 ('Mauritius', 2020, Decimal('100.00'), Decimal('96.00'), Decimal('1.265740'), Decimal('11.40')),
 ('Mayotte', 2020, Decimal('96.00'), Decimal('100.00'), None, None),
 ('Mozambique', 2020, Decimal('66.67'), Decimal('40.33'), Decimal('31.178239'), Decimal('14.03')),
 ('Rwanda', 2020, Decimal('66.33'), Decimal('64.00'), Decimal('13.146362'), Decimal('10.18')),
 ('Réunion', 2020, Decimal('100.00'), Decimal('100.00'), None, None),
 ('Somalia', 2020, Decimal('57.33'), Decimal('40.00'), Decimal('16.537016'), Decimal('6.88')),
 ('South Sudan', 2020, Decimal('48.33'), Decimal('22.33'), Decimal('10.606227'), None),
 ('Uganda', 2020, Decimal('61.00'), Decimal('21.67'), Decimal('44.404611'), Decimal('37.60')),
 ('United Republic of Tanzania', 2020, Decimal('65.00'), Decimal('34.00'), None, None),
 ('Zambia', 2020, Decimal('66.67'), Decimal('32.67'), Decimal('18.927715'), Decimal('18.11')),
 ('Zimbabwe', 2020, Decimal('68.00'), Decimal('36.33'), Decimal('15.669666'), Decimal('21.51')),
 ('Angola', 2020, Decimal('52.33'), Decimal('47.00'), Decimal('33.428486'), Decimal('53.62')),
 ('Cameroon', 2020, Decimal('64.00'), Decimal('43.00'), Decimal('26.491087'), Decimal('40.77')),
 ('Central African Republic', 2020, Decimal('38.33'), Decimal('15.00'), Decimal('5.343020'), Decimal('2.33')),
 ('Chad', 2020, Decimal('52.67'), Decimal('18.67'), Decimal('16.644701'), Decimal('10.72')),
 ('Congo', 2020, Decimal('69.00'), Decimal('17.67'), None, None),
 ('Democratic Republic of the Congo', 2020, Decimal('47.67'), Decimal('15.33'), None, None),
 ('Gabon', 2020, Decimal('73.33'), Decimal('47.00'), Decimal('2.292573'), Decimal('15.31')),
 ('Sao Tome and Principe', 2020, Decimal('77.33'), Decimal('46.00'), Decimal('0.218641'), Decimal('0.47')),
 ('Botswana', 2020, Decimal('89.67'), Decimal('74.33'), Decimal('2.546402'), Decimal('14.93')),
 ('Eswatini', 2020, Decimal('76.67'), Decimal('61.33'), Decimal('1.180655'), Decimal('3.98')),
 ('Lesotho', 2020, Decimal('76.33'), Decimal('49.67'), Decimal('2.254100'), Decimal('2.23')),
 ('Namibia', 2020, Decimal('83.67'), Decimal('35.00'), Decimal('2.489098'), Decimal('10.58')),
 ('South Africa', 2020, Decimal('92.00'), Decimal('78.67'), Decimal('58.801927'), Decimal('337.62')),
 ('Benin', 2020, Decimal('65.33'), Decimal('17.33'), Decimal('12.643123'), Decimal('15.65')),
 ('Burkina Faso', 2020, Decimal('53.33'), Decimal('25.00'), Decimal('21.522626'), Decimal('17.93')),
 ('Cabo Verde', 2020, Decimal('87.33'), Decimal('78.00'), Decimal('0.582640'), Decimal('1.70')),
 ("Côte d'Ivoire", 2020, Decimal('70.67'), Decimal('34.67'), None, None),
 ('Gambia', 2020, Decimal('79.33'), Decimal('44.33'), None, None),
 ('Ghana', 2020, Decimal('84.67'), Decimal('23.00'), Decimal('32.180401'), Decimal('70.04')),
 ('Guinea', 2020, Decimal('67.33'), Decimal('32.33'), Decimal('13.205153'), Decimal('14.18')),
 ('Guinea-Bissau', 2020, Decimal('60.00'), Decimal('19.33'), Decimal('2.015828'), Decimal('1.43')),
 ('Liberia', 2020, Decimal('75.00'), Decimal('17.67'), Decimal('5.087584'), Decimal('3.04')),
 ('Mali', 2020, Decimal('83.67'), Decimal('46.00'), Decimal('21.224040'), Decimal('17.47')),
 ('Mauritania', 2020, Decimal('70.33'), Decimal('48.00'), Decimal('4.498604'), Decimal('8.41')),
 ('Niger', 2020, Decimal('57.33'), Decimal('24.67'), Decimal('24.333639'), Decimal('13.74')),
 ('Nigeria', 2020, Decimal('77.33'), Decimal('42.67'), Decimal('208.327405'), Decimal('432.20')

### Task 2

In [48]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    region
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND 
	Est_gdp_in_billions IS NULL;

# Use LIMIT if you think the results set will be large



 * mysql+pymysql://root:***@localhost:3306/united_nations
9 rows affected.
0 rows affected.


[]

### Task 3

In [49]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    	region
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND 
	Est_gdp_in_billions IS NOT NULL;

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results

 * mysql+pymysql://root:***@localhost:3306/united_nations
38 rows affected.
0 rows affected.


[]

### Task 4

Arrange the data in descending order. Comparing Nigeria to South Africa.  We see that Nigeria has a lower percentage in the availability of both water and sanitation services. 


In [55]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services

WHERE
	Region = 'Sub-Saharan Africa'
AND 
Time_period = 2020

AND 
	Est_gdp_in_billions IS NOT NULL

AND 
	Country_name  NOT IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana');


# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results


 * mysql+pymysql://root:***@localhost:3306/united_nations
33 rows affected.
0 rows affected.


[]

### Task 5

Looking at the first entry without ordering any data it is interesting to note that Burundi, with a GDP of only 2.65 Billion has a similar percentage of managed services compared to Nigeria seen in our previous  query(77% and 42%). Nigeria had a percentage of 77 for managed water and 42 for managed sanitation. Arranging the percentage managed drinking water services in descending order.Mauritius with a GDP lower than that of Botswana has better drinking water and sanitation services. 

In [54]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services

WHERE
	Region = 'Sub-Saharan Africa'
AND 
Time_period = 2020

AND 
	Est_gdp_in_billions IS NOT NULL

AND 
	Country_name in ('Nigeria','South Africa','Ethiopia','Kenya','Ghana');

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results by GDP
# Use ORDER BY Pct_managed_drinking_water_services to order percentage managed drinking water services
# Use ORDER BY Pct_managed_sanitation_services to order your results by percentage managed sanitation services


 * mysql+pymysql://root:***@localhost:3306/united_nations
5 rows affected.
0 rows affected.


[]

## Summary

In this exercise we used the IS NULL statement to determine if there were any null values in the GDP column. 
We then used the IS NOT NULL statement to exclude those nulls.
We looked at the top 5 GDP's in Sub-Saharan Africa by using the IS IN statement. 
We had a look at the rest of the Sub-Saharan African countries by excluding the top 5 GDP's using the IS NOT IN statement
Based on the quick perusal of the data in Sub-Saharan africa, we can conclude that there isn’t
any noticeable correlation between GDP and the availability of drinking water and sanitation services. 


#  

<div align="center" style=" font-size: 80%; text-align: center; margin: 0 auto">
<img src="https://raw.githubusercontent.com/Explore-AI/Pictures/master/ExploreAI_logos/EAI_Blue_Dark.png"  style="width:200px";/>
</div>